lib/web/apiserver.go

```diff
@@ -2142,8 +2142,56 @@ func (h *Handler) siteNodeConnect(
 		return nil, trace.Wrap(err)
 	}
 
+	// Fetching nodes is quite slow when many nodes exist so fetch it once and
+	// pass the value to the methods that require it.
+	clt, err := ctx.GetUserClient(site)
```

Comment:
While we're here, I think we should rename `ctx` to `sctx`. Any variable in go named `ctx` that is not of type `context.Context` really throws me off. It's like having a variable named `err` that is not an `error`.

Reply:
good call, done.